Cargo.toml

```diff
@@ -56,74 +57,89 @@ primitives = { path = "primitives", default-features = false }
-frame-benchmarking = { branch = "polkadot-v0.9.40", git = "https://github.com/paritytech/substrate.git", default-features = false }
-frame-executive = { branch = "polkadot-v0.9.40", git = "https://github.com/paritytech/substrate.git", default-features = false }
-frame-election-provider-support = { branch = "polkadot-v0.9.40", git = "https://github.com/paritytech/substrate.git", default-features = false }
-frame-support = { branch = "polkadot-v0.9.40", git = "https://github.com/paritytech/substrate.git", default-features = false }
-frame-system = { branch = "polkadot-v0.9.40", git = "https://github.com/paritytech/substrate.git", default-features = false }
-frame-system-benchmarking = { branch = "polkadot-v0.9.40", git = "https://github.com/paritytech/substrate.git", default-features = false }
-frame-system-rpc-runtime-api = { branch = "polkadot-v0.9.40", git = "https://github.com/paritytech/substrate.git", default-features = false }
-pallet-balances = { branch = "polkadot-v0.9.40", git = "https://github.com/paritytech/substrate.git", default-features = false }
-pallet-insecure-randomness-collective-flip = { branch = "polkadot-v0.9.40", git = "https://github.com/paritytech/substrate.git", default-features = false }
-pallet-session = { branch = "polkadot-v0.9.40", git = "https://github.com/paritytech/substrate.git", default-features = false }
-pallet-staking-substrate = { package = "pallet-staking", branch = "polkadot-v0.9.40", git = "https://github.com/paritytech/substrate.git", default-features = false }
-pallet-scheduler = { branch = "polkadot-v0.9.40", git = "https://github.com/paritytech/substrate.git", default-features = false }
-pallet-sudo = { branch = "polkadot-v0.9.40", git = "https://github.com/paritytech/substrate.git", default-features = false }
-pallet-timestamp = { branch = "polkadot-v0.9.40", git = "https://github.com/paritytech/substrate.git", default-features = false }
-pallet-transaction-payment = { branch = "polkadot-v0.9.40", git = "https://github.com/paritytech/substrate.git", default-features = false }
-pallet-transaction-payment-rpc = { branch = "polkadot-v0.9.40", git = "https://github.com/paritytech/substrate.git" }
-pallet-transaction-payment-rpc-runtime-api = { branch = "polkadot-v0.9.40", git = "https://github.com/paritytech/substrate.git", default-features = false }
-sc-basic-authorship = { branch = "polkadot-v0.9.40", git = "https://github.com/paritytech/substrate.git" }
-sc-chain-spec = { branch = "polkadot-v0.9.40", git = "https://github.com/paritytech/substrate.git" }
-sc-cli = { branch = "polkadot-v0.9.40", git = "https://github.com/paritytech/substrate.git" }
-sc-client-api = { branch = "polkadot-v0.9.40", git = "https://github.com/paritytech/substrate.git" }
-sc-client-db = { branch = "polkadot-v0.9.40", git = "https://github.com/paritytech/substrate.git", features = [
+frame-benchmarking-cli = { branch = "pos-keep-history", git = "https://github.com/gluwa/substrate.git" }
```

Comment:
I'd prefer if the `pos-keep-history` branch contains an explicit version name and is created from the relevant polkadot-v* branch. For example:
- `pos-keep-history-polkadot-v0.9.40` is based on upstream `polkadot-v0.9.40` branch
- `pos-keep-history-polkadot-v0.9.41` is based on upstream `polkadot-v0.9.41` branch

and so on. 

This will make versioning explicit and also allow us to "rebase" the `pos-keep-history` branches independently of Creditcoin to facilitate easier dependency upgrades.

Reply:
The current `pos-keep-history` branch seems to be based on `polkadot-v0.9.40` but we have `polkadot-v0.9.41` already available (no changes required to Creditcoin, see #1029).  We'll need at least v0.9.42 before we can drop the nightly compiler.

Do you mind forking off of `polkadot-v0.9.41`  as mentioned above ?